engine-tests/src/main/java/org/terasology/HeadlessEnvironment.java

```diff
@@ -202,15 +188,15 @@ protected AssetManager setupAssetManager() {
 
         // cast lambdas explicitly to avoid inconsistent compiler behavior wrt. type inference
         assetTypeManager.registerCoreAssetType(Prefab.class,
-                (AssetFactory<Prefab, PrefabData>) PojoPrefab::new, false, "prefabs");
+                PojoPrefab::new, false, "prefabs");
```

Comment:
This is a nice cleanup, but is it really needed as part of the feature? It bloats the diff for something that is already quite an undertaking :-)

Not saying to revert these if they are benign - it can just be a good practice to separate code style improvements from logical changes when you're doing anything of substantial size.